lib/awful/tag.lua

```diff
@@ -284,7 +283,7 @@ function tag.object.delete(self, fallback_tag, force)
 
     -- Abort if we would have un-tagged clients.
     local clients = self:clients()
-    if ( #clients > 0 and ntags <= 1 ) or fallback_tag == nil then return end
+    if #clients > 0 and fallback_tag == nil then return false end
```

Comment:
> The old logic was "don't allow deleting the last tag of a screen". I think that's quite a good idea since screens without tags will surely cause many problems.

The old logic was inconsistent with Awesome default state. No tags is currently supported and is heavily used by the Shifty extension. By default, Shifty has no tags and they are generated on the fly. This logic bug caused it to be unable to return to its original state. Tyrannical is also mildly affected by this (it isn't the default, but this behavior can be defined). But mostly, it bugs when you remove the screens as `:delete()` doesn't do its job anymore.

 > (By the way, did you change the or in the old code into an and on purpose?)

Yes,  returning early prevented important `:delete()` logic from being executed. When the tag _has_ to be deleted, then it _has_ to be deleted. Removing screens is a good example where the `:delete()` method has to work. Anyway, if `#clients > 0` is true, it still return, the fallback tag is irrelevant if there is nothing looking for a fallback, the previous code was wrong.